### PR TITLE
feat: add polkadot-launch config

### LINKED ISSUE #42: Failed to produce block on local rococo testnet
## Describe the bug

|Test Machine|
|-|
|OSX Big Sur 11.2.3 (M1 chip)|

PINT can not produce block after being registered as a parachain on the local rococo testnet

## Expected Behavior

Follow the guide of [substrate parachain template][template], PINT should be able to print logs like

```  
2021-04-01 16:31:09 [Relaychain] 💤 Idle (2 peers), best: #243 (0x46d8…f394), finalized #192 (0x9fb4…4b28), ⬇ 1.0kiB/s ⬆ 3.2kiB/s    
2021-04-01 16:31:09 [Parachain] 💤 Idle (0 peers), best: #90 (0x85c6…45be), finalized #64 (0x10af…4ede), ⬇ 1.1kiB/s ⬆ 1.0kiB/s    
2021-04-01 16:31:12 [Relaychain] ✨ Imported #244 (0xe861…d99d)    
2021-04-01 16:31:14 [Relaychain] 💤 Idle (2 peers), best: #244 (0xe861…d99d), finalized #193 (0x9225…85f1), ⬇ 2.0kiB/s ⬆ 1.6kiB/s    
2021-04-01 16:31:14 [Parachain] 💤 Idle (0 peers), best: #90 (0x85c6…45be), finalized #65 (0xdd20…d44a), ⬇ 1.6kiB/s ⬆ 1.4kiB/s    
```

Notice here is `...best: #90..., finalized #65...`

## Current Behavior

```
2021-04-22 21:03:24  [Relaychain] 💤 Idle (2 peers), best: #295 (0xd339…84bc), finalized #292 (0xb3eb…aed3), ⬇ 2.3kiB/s ⬆ 2.0kiB/s    
2021-04-22 21:03:24  [Parachain] 💤 Idle (0 peers), best: #0 (0xd0be…fc70), finalized #0 (0xd0be…fc70), ⬇ 2.3kiB/s ⬆ 2.0kiB/s    
2021-04-22 21:03:29  [Relaychain] 💤 Idle (2 peers), best: #295 (0xd339…84bc), finalized #293 (0x2799…efa4), ⬇ 1.0kiB/s ⬆ 0.8kiB/s    
2021-04-22 21:03:29  [Parachain] 💤 Idle (0 peers), best: #0 (0xd0be…fc70), finalized #0 (0xd0be…fc70), ⬇ 0 ⬆ 0 
```

`...best: #0..., finalized #0...`

## Possible Solution

Check the runtime configuration of PINT.

## To Reproduce

Just follow the [substrate-parachain-template][template], but replace the template with PINT.

[template]: https://github.com/substrate-developer-hub/substrate-parachain-template#registering-on-local-relay-chain

### PR DESCRIPTION
**Pull request by mattsse**
## Changes

- Uses `pint` naming within the  cli and chain-spec
- Add [polkadot-launch](https://github.com/paritytech/polkadot-launch) config file 

## Tests

At least on my machine the master branch version of `polkadot-launch` failed to wait until the nodes are spawned. I used the [polkadot-launch/wait-for-port](https://github.com/mattsse/polkadot-launch/tree/wait-for-port) branch with a hotfix that waits until the websocket port of the node is available

```
git clone -b wait-for-port https://github.com/mattsse/polkadot-launch
cd polkadot-launch
yarn install
yarn run start <path to PINT>/config.json 
```

## Issues

- Closes #42

### CODE REVIEW

File: node/Cargo.toml [hunk 1]
```diff
@@ -202,7 +202,7 @@ git = "https://github.com/paritytech/polkadot"
 branch = 'rococo-v1'
 
 [[bin]]
-name = 'parachain-collator'
+name = 'pint'
```
Comment (clearloop) [inline node/Cargo.toml:205]: We can just delete this 

```toml
[[bin]]
name = "pint"
```

As we already have

```toml
[package]
name = "pint"
```

**Review by clearloop — COMMENTED** [review]
Follow the tests above, I'm stuck at 

```
 𝝺 yarn run start ../PINT/config.json
yarn run v1.22.10
$ yarn build && node dist/index.js ../PINT/config.json
$ tsc
2021-04-23 20:48:07  Building chain spec    
Starting with a fresh authority set:
Added Authority alice
Added Authority bob
Added Authority charlie
2021-04-23 20:48:08  Building chain spec    
2021-04-23 20:48:09  Took active validators from set with wrong size    
2021-04-23 20:48:09  Took active validators from set with wrong size    
Starting alice...
Starting bob...
Starting charlie...
Waiting for localhost:9944........
Connected!
2021-04-23 20:48:17        REGISTRY: Unable to resolve type Outcome, it will fail on construction
2021-04-23 20:48:17        API/INIT: RPC methods not decorated: beefy_subscribeJustifications, beefy_unsubscribeJustifications
Starting a Collator for parachain 1: 5Ec4AhNmTPsRSiBHeyHEMojCJFbKn7aoT3KNKnQEc3gV3ocB, Collator port : 31200 wsPort : 9988
Added --chain=pint-dev
Added  to parachain
Added --execution=wasm to collator
```

And checking the served `:9944` polkadot.app UI

Screen Shot 2021-04-23 at 8 52 49 PM

PINT hasn't been registered as a parachain, and the issue #42 mentioned is that after PINT has been registered as a parachin, it doesn't produce blocks, can your paste the logs or screenshot PINT produces blocks on your machine @mattsse , and I will try to find out the problem my machine has

File: config.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	"hrmpChannels": [
+		{
+			"sender": 1,
+			"recipient": 300,
+			"maxCapacity": 8,
+			"maxMessageSize": 512
+		}
+	],
+	"types": {},
+	"finalization": false
```
Comment (clearloop) [inline config.json:54]: Why `finalization: false`?

**Comment by mattsse** [comment]
Did you build the correct polkadot branch `rococo-v1` and put it into `bin` as described [here](https://github.com/paritytech/polkadot-launch#binary-files)? (you only need polkadot)

**Comment by clearloop** [comment]
> Did you build the correct polkadot branch `rococo-v1` and put it into `bin` as described [here](https://github.com/paritytech/polkadot-launch#binary-files)? (you only need polkadot)

```
 𝝺 git branch
  master
  rococo-branch
* rococo-v1
 𝝺 cargo build --release --features=real-overseer
error: Package `polkadot v0.8.30 (/Users/clearloop/code/polkadot)` does not have the feature `real-overseer`
```

I tested the integration with a binary built by `cargo build --release` on `rococo-v1` branch yesterday...And it seems I can't `cargo build --release --features=real-overseer` following the `#binary-files` LOL

**Comment by clearloop** [comment]
Just dismissed the `9988.log` on my last review,

```
 𝝺 tail -f 9988.log 
Error: Input("Error opening spec file: No such file or directory (os error 2)")
```

Let me test it again

**Comment by mattsse** [comment]
I think the `real-overseer` feature was removed recently and isn't required anymore. I've built it with `cargo build --release`

File: config.json (hunk 2 above)
Comment (mattsse) [reply config.json:54]: This flag is described here https://github.com/paritytech/polkadot-launch#finalization, setting it to `false` should speed things up

**Review by clearloop — APPROVED** [review]
```
2021-04-23 21:43:33 [Parachain] 💤 Idle (0 peers), best: #1 (0xd4ef…2af9), finalized #0 (0xb329…fd7c), ⬇ 57 B/s ⬆ 55 B/s    
2021-04-23 21:43:33 [Relaychain] 💤 Idle (4 peers), best: #24 (0x665a…e7f9), finalized #21 (0xa6b1…c98c), ⬇ 1.3kiB/s ⬆ 1.0kiB/s    
2021-04-23 21:43:36 [Relaychain] ✨ Imported #25 (0xf729…d9f4)    
2021-04-23 21:43:38 [Parachain] 💤 Idle (0 peers), best: #1 (0xd4ef…2af9), finalized #1 (0xd4ef…2af9), ⬇ 0 ⬆ 0    
2021-04-23 21:43:38 [Relaychain] 💤 Idle (4 peers), best: #25 (0xf729…d9f4), finalized #23 (0x6c72…d4c3), ⬇ 1.7kiB/s ⬆ 1.0kiB/s    
2021-04-23 21:43:42 [Relaychain] ✨ Imported #26 (0xd932…c568)  
```

It works on my machine now! Even I am confused about how this `wait-for-port` can solve this problem but this is really amazing!!!

BTW, it would be nice renaming `PINT/config.json` to `PINT/polkadot-launch.json`.

File: node/Cargo.toml (hunk 1 above)
Comment (mattsse) [reply node/Cargo.toml:205]: Good catch

**Comment by mattsse** [comment]
> BTW, it would be nice renaming PINT/config.json to PINT/polkadot-launch.json.

The `polkadot-launch` always refers to it as the configuration file and uses `config.json` consistently, so I think we should also stick to `config.json`

**Pull request merged**